api/traces/v1/trace_rbac.go

```diff
@@ -0,0 +1,227 @@
+package v1
+
+import (
+	"bytes"
+	"compress/flate"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	// nolint:staticcheck
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/grafana/tempo/pkg/tempopb"
+	commonv1 "github.com/grafana/tempo/pkg/tempopb/common/v1"
+	tracev1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+
+	apilogsv1 "github.com/observatorium/api/api/logs/v1"
+)
+
+const (
+	namespaceAttributeKey = "k8s.namespace.name"
+	serviceAttributeKey   = "service.name"
+)
+
+func WithTraceQLNamespaceSelect() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if !strings.Contains(r.URL.Path, "/api/search") {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			query := r.URL.Query()
+			q := query.Get("q")
+			traceQL, err := url.QueryUnescape(q)
+			if err != nil {
+				next.ServeHTTP(w, r)
+				return
+			}
+			if traceQL == "" {
+				traceQL = "{ }"
+			}
+			traceQL = traceQL + " | select(resource.k8s.namespace.name)"
+			query.Set("q", traceQL)
+			r.URL.RawQuery = query.Encode()
+			next.ServeHTTP(w, r)
+		})
+	}
+}
+
+func responseRBACModifier(log log.Logger) func(response *http.Response) error {
+	return func(response *http.Response) error {
+
+		allowedNamespaces := map[string]bool{}
+		namespaces := apilogsv1.AllowedNamespaces(response.Request.Context())
+		for _, ns := range namespaces {
+			allowedNamespaces[ns] = true
+		}
+		level.Debug(log).Log("AllowedNamespaces", allowedNamespaces)
+
+		if strings.Contains(response.Request.URL.Path, "/api/traces/") || strings.Contains(response.Request.URL.Path, "/api/search") {
```

Comment:
Should we send a 403 error for other API endpoints?
https://grafana.com/docs/tempo/latest/api_docs/#endpoints

https://grafana.com/docs/tempo/latest/api_docs/#search-tag-values-v2 could leak private data (but this will also disable the auto-complete in the tracing UI :/)

Reply:
That is a great point. 

What endpoints is using the UI? 

Another iteration on this feature will be supporting a white listing of attributes. Then we could enable it and return only allowed ones.

Reply:
The UI uses
```
/api/traces
/api/search
/api/v2/search/tags
/api/v2/search/tag
```

The last 2 are for auto-completion of attribute names and values.

Reply:
Added the 403 check.